pkg/mounter/safe_mounter_windows.go

```diff
@@ -0,0 +1,264 @@
+		LocalPath:  normalizeWindowsPath(target),
+		RemotePath: source,
+		Username:   options[0],
+		Password:   options[1],
+	}
+	_, err = mounter.SMBClient.NewSmbGlobalMapping(context.Background(), smbMountRequest)
+	if err != nil {
+		return fmt.Errorf("smb mapping failed with error: %v", err)
+	}
+	return nil
+}
+
+func (mounter *CSIProxyMounter) SMBUnmount(target string) error {
+	klog.V(4).Infof("SMBUnmount: local path: %s", target)
+	// TODO: We need to remove the SMB mapping. The change to remove the
+	// directory brings the CSI code in parity with the in-tree.
+	return mounter.Rmdir(target)
+}
+
+// Mount just creates a soft link at target pointing to source.
+func (mounter *CSIProxyMounter) Mount(source string, target string, fstype string, options []string) error {
+	klog.V(4).Infof("Mount: old name: %s. new name: %s", source, target)
+	// Mount is called after the format is done.
+	// TODO: Confirm that fstype is empty.
+	linkRequest := &fsv1alpha1.LinkPathRequest{
+		SourcePath: normalizeWindowsPath(source),
+		TargetPath: normalizeWindowsPath(target),
+	}
+	_, err := mounter.FsClient.LinkPath(context.Background(), linkRequest)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Rmdir - delete the given directory
+// TODO: Call separate rmdir for pod context and plugin context. v1alpha1 for CSI
+//       proxy does a relaxed check for prefix as c:\var\lib\kubelet, so we can do
+//       rmdir with either pod or plugin context.
+func (mounter *CSIProxyMounter) Rmdir(path string) error {
+	klog.V(4).Infof("Remove directory: %s", path)
+	rmdirRequest := &fsv1alpha1.RmdirRequest{
+		Path:    normalizeWindowsPath(path),
+		Context: fsv1alpha1.PathContext_POD,
+		Force:   true,
+	}
+	_, err := mounter.FsClient.Rmdir(context.Background(), rmdirRequest)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Unmount - Removes the directory - equivalent to unmount on Linux.
+func (mounter *CSIProxyMounter) Unmount(target string) error {
+	klog.V(4).Infof("Unmount: %s", target)
+	return mounter.Rmdir(target)
+}
+
+func (mounter *CSIProxyMounter) List() ([]mount.MountPoint, error) {
+	return []mount.MountPoint{}, fmt.Errorf("List not implemented for CSIProxyMounter")
+}
+
+func (mounter *CSIProxyMounter) IsMountPointMatch(mp mount.MountPoint, dir string) bool {
+	return mp.Path == dir
+}
+
+// IsLikelyMountPoint - If the directory does not exists, the function will return os.ErrNotExist error.
+//   If the path exists, call to CSI proxy will check if its a link, if its a link then existence of target
+//   path is checked.
+func (mounter *CSIProxyMounter) IsLikelyNotMountPoint(path string) (bool, error) {
+	klog.V(4).Infof("IsLikelyNotMountPoint: %s", path)
+	isExists, err := mounter.ExistsPath(path)
+	if err != nil {
+		klog.Error(err)
+		return false, err
+	}
+	if !isExists {
+		klog.Error("returning not exists")
+		return true, os.ErrNotExist
+	}
+
+	response, err := mounter.FsClient.IsMountPoint(context.Background(),
```

Comment:
Do we need to handle the response.Error？Is it equal to the other return value `err`?

Reply:
Initially the APIs had both error outside and error within. We decided to pass the errors only via the outside one. Will be removing those internal error in alpha2 version of the APIs.

Reply:
Got it.